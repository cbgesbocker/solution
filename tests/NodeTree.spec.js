const NodeTree = require("../src/lib/NodeTree");

describe("NodeTree.js", () => {
  let ntree, getNodeSpy;
  beforeEach(async () => {
    getNodeSpy = jest.fn(async (id) => ({
      data: [
        {
          id,
          child_node_ids: [],
        },
      ],
    }));
    ntree = new NodeTree(
      {
        id: "test-node",
        child_node_ids: ["1234", "12345"],
      },
      { getNode: getNodeSpy }
    );
  });

  test("Collect tree data should call getNode function", async () => {
    await ntree.collectTreeData();
    expect(getNodeSpy).toHaveBeenCalled();
  });

  test("Collect tree data should call getNode function with the right # of times", async () => {
    await ntree.collectTreeData();
    expect(getNodeSpy).toHaveBeenCalledTimes(2);
  });

  test("Collect tree data should setup the object metadata for data collection properly", async () => {
    await ntree.collectTreeData();
    expect(ntree).toMatchSnapshot();
  });
});
