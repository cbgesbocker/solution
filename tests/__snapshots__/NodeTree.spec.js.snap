// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NodeTree.js Collect tree data should setup the object metadata for data collection properly 1`] = `
NodeTree {
  "getNode": [MockFunction] {
    "calls": Array [
      Array [
        "1234",
      ],
      Array [
        "12345",
      ],
    ],
    "results": Array [
      Object {
        "type": "return",
        "value": Promise {},
      },
      Object {
        "type": "return",
        "value": Promise {},
      },
    ],
  },
  "max": Object {
    "count": 1,
    "node": Object {
      "child_node_ids": Array [],
      "id": "1234",
    },
  },
  "nodeCountDict": Object {
    "1234": 1,
    "12345": 1,
    "test-node": 1,
  },
  "nodeDict": Object {
    "1234": Object {
      "child_node_ids": Array [],
      "id": "1234",
    },
    "12345": Object {
      "child_node_ids": Array [],
      "id": "12345",
    },
    "test-node": Object {
      "child_node_ids": Array [
        "1234",
        "12345",
      ],
      "id": "test-node",
    },
  },
  "rootNode": Object {
    "child_node_ids": Array [
      "1234",
      "12345",
    ],
    "id": "test-node",
  },
  "visited": Array [
    "test-node",
    "1234",
    "12345",
  ],
}
`;
